morpheus/_lib/src/io/loaders/rest.cpp

```diff
@@ -17,21 +17,238 @@
+                                         py::module_& mod_cudf,
+                                         const std::string& strategy)
+{
+    std::string df_json_str = beast::buffers_to_string(response.body().data());
+    boost::algorithm::trim(df_json_str);
+
+    // When calling cudf.read_json() with engine='cudf', it expects an array object as input.
+    // The workaround here is to add square brackets if the original data is not represented as an array.
+    // Submitted an issue to cudf: https://github.com/rapidsai/cudf/issues/13527.
+    if (!(df_json_str.front() == '[' && df_json_str.back() == ']'))
+    {
+        df_json_str = "[" + df_json_str + "]";
+    }
+
+    py::gil_scoped_acquire gil;
+    auto current_df = mod_cudf.attr("DataFrame")();
+    current_df      = mod_cudf.attr("read_json")(py::str(df_json_str), py::str("cudf"));
+    if (dataframe.is_none())
+    {
+        dataframe = current_df;
+    }
+    else if (strategy == "aggregate")
+    {
+        py::list args;
+        args.attr("append")(dataframe);
+        args.attr("append")(current_df);
+        dataframe = mod_cudf.attr("concat")(args);
+    }
+}
+
 std::shared_ptr<ControlMessage> RESTDataLoader::load(std::shared_ptr<ControlMessage> message, nlohmann::json task)
 {
     VLOG(30) << "Called RESTDataLoader::load()";
 
-    // TODO(Devin): Implement this
-    throw std::runtime_error("RESTDataLoader::load() not implemented yet");
+    // Aggregate dataframes for each file
+    py::gil_scoped_acquire gil;
+    py::module_ mod_cudf;
+    py::object dataframe = py::none();
+
+    auto& cache_handle = mrc::pymrc::PythonObjectCache::get_handle();
+    mod_cudf           = cache_handle.get_module("cudf");
+    py::gil_scoped_release release;
+
+    // TODO(Devin) : error checking + improve robustness
+    if (!task["queries"].is_array() or task.empty())
+    {
+        throw std::runtime_error("'REST Loader' control message specified no queries to load");
+    }
+
+    std::string strategy = task.value("strategy", "aggregate");
+    if (strategy != "aggregate")
+    {
+        throw std::runtime_error("Only 'aggregate' strategy is currently supported");
+    }
+
+    auto conf     = this->config();
+    int max_retry = conf.value("max_retry", 3);
+    auto queries  = task["queries"];
+    // TODO(Devin) : Migrate this to use the cudf::io interface
+    for (auto& query : queries)
+    {
+        std::string method = query.value("method", "GET");
+        std::transform(method.begin(), method.end(), method.begin(), ::toupper);
+        std::string endpoint = query.value("endpoint", "");
+        if (endpoint.empty())
+        {
+            continue;
+        }
+        std::string content_type = query.value("content_type", "");
+        std::string body         = query.value("body", "");
+        nlohmann::json x_headers = query.value("x-headers", nlohmann::json());
+        std::unordered_map<std::string, std::string> x_headers_map;
```

Comment:
Since there is a decent amount of work going on here, it will probably be worth it to make this section more declarative as well.